perception/traffic_light_recognition/traffic_light_classifier/src/nodelet.cpp

```diff
@@ -18,55 +18,58 @@
 
 namespace traffic_light
 {
-void TrafficLightClassifierNodelet::onInit()
+TrafficLightClassifierNodelet::TrafficLightClassifierNodelet(const rclcpp::NodeOptions & options)
+: Node("traffic_light_classifier_node", options)
 {
-  nh_ = getNodeHandle();
-  pnh_ = getPrivateNodeHandle();
-  image_transport_.reset(new image_transport::ImageTransport(pnh_));
-  pnh_.param<bool>("approximate_sync", is_approximate_sync_, false);
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+  is_approximate_sync_ = this->declare_parameter("approximate_sync", false);
   if (is_approximate_sync_) {
     approximate_sync_.reset(new ApproximateSync(ApproximateSyncPolicy(10), image_sub_, roi_sub_));
     approximate_sync_->registerCallback(
-      boost::bind(&TrafficLightClassifierNodelet::imageRoiCallback, this, _1, _2));
+      std::bind(&TrafficLightClassifierNodelet::imageRoiCallback, this, _1, _2));
   } else {
     sync_.reset(new Sync(SyncPolicy(10), image_sub_, roi_sub_));
     sync_->registerCallback(
-      boost::bind(&TrafficLightClassifierNodelet::imageRoiCallback, this, _1, _2));
+      std::bind(&TrafficLightClassifierNodelet::imageRoiCallback, this, _1, _2));
   }
-  ros::SubscriberStatusCallback connect_cb = boost::bind(&TrafficLightClassifierNodelet::connectCb, this);
-  std::lock_guard<std::mutex> lock(connect_mutex_);
-  tl_states_pub_ = pnh_.advertise<autoware_perception_msgs::TrafficLightStateArray>(
-    "output/traffic_light_states", 1, connect_cb, connect_cb);
 
-  int classifier_type;
-  pnh_.param<int>(
-    "classifier_type", classifier_type, TrafficLightClassifierNodelet::ClassifierType::HSVFilter);
+  image_sub_.subscribe(this, "input/image", "raw", rclcpp::QoS{1}.get_rmw_qos_profile());
+  roi_sub_.subscribe(this, "input/rois", rclcpp::QoS{1}.get_rmw_qos_profile());
+
+  tl_states_pub_ = this->create_publisher<autoware_perception_msgs::msg::TrafficLightStateArray>(
+    "output/traffic_light_states", rclcpp::QoS{1});
+
+  int classifier_type = this->declare_parameter(
+    "classifier_type", static_cast<int>(TrafficLightClassifierNodelet::ClassifierType::HSVFilter));
   if (classifier_type == TrafficLightClassifierNodelet::ClassifierType::HSVFilter) {
-    classifier_ptr_ = std::make_shared<ColorClassifier>(nh_, pnh_);
+    classifier_ptr_ = std::make_shared<ColorClassifier>(this);
```

Comment:
If you prefer a `shared_ptr` (I don't think it's necessary since the `classifier_ptr_` should never outlive the main node anyway), you could use `shared_from_this()`.

Reply:
I don't mind either way, but I think we can keep it as normal pointer for this use case for now.